File: ServerSpace/src/main/java/Business/InfraLista.java
package Business;

import EntityLayer.InfraListaEntity;
import Framework.Inj;

import java.util.ArrayList;

import DataLayer.InfraListaDB;

public class InfraLista {
    
  
    public ArrayList<InfraListaEntity> ObtenerCodigo(String codigo) {
        InfraListaDB BD = new InfraListaDB();
        return BD.ObtenerCodigo(codigo);
    }

    public ArrayList<InfraListaEntity> Buscar(String codigo, String Nombre) {
        InfraListaDB BD = new InfraListaDB();
        return BD.Buscar(codigo, Nombre);
    }

    public ArrayList<InfraListaEntity> ObtenerMain(String codigo) {
        InfraListaDB BD = new InfraListaDB();
        return BD.ObtenerMain(codigo);
    }

    public InfraListaEntity Registrar(InfraListaEntity Item) {
        try {
            Inj.IniciarTranssaccion(true);
            InfraListaDB BD = new InfraListaDB();
            BD.Registrar(Item);
            Inj.FinalizarTranssaccion();
            return Item;
        } catch (Exception ex) {
            System.out.println("ERROR " + ex);
            throw new UnsupportedOperationException("MENSAJE :" + ex);
        }
    }

    public ArrayList<InfraListaEntity> ObtenerItem(int ListaId) {
        InfraListaDB BD = new InfraListaDB();
        return BD.ObtenerItem(ListaId);
    }
}

File: Space/Space.BusinessLayer/InfraLista.cs
using Framework;
using Space.DataLayer;
using Space.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Space.BusinessLayer
{
    public class InfraLista
    {

  

     

        public static List<InfraListaEntity> ObtenerItemsCodigo(String codigo)
        {
            try
            {
                InfraListaDB DB = new InfraListaDB();
                return DB.ObtenerItemsCodigo(codigo);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        public static List<InfraListaEntity> BuscarItemCodigo(String codigo, String Nombre)
        {
            try
            {
                InfraListaDB DB = new InfraListaDB();
                return DB.BuscarItemCodigo(codigo, Nombre);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        public static List<InfraListaEntity> ObtenerMain(String codigo)
        {
            try
            {
                InfraListaDB DB = new InfraListaDB();
                return DB.ObtenerMain(codigo);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        public static Int32 Registrar(InfraListaEntity Ent)
        {
            try
            {
                InfraListaDB DB = new InfraListaDB();
                DB.Save(Ent);
                return Ent.ListaId;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }



        public static List<InfraListaEntity> ObtenerItem(Int32 ListaId)
        {
            try
            {
                InfraListaDB DB = new InfraListaDB();
                return DB.ObtenerItem(ListaId);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        public static List<InfraListaEntity> ObtenerEnlaceItem(Int32 ListaId)
        {
            try
            {
                InfraListaDB DB = new InfraListaDB();
                return DB.ObtenerEnlaceItem(ListaId);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }


    }
}
